src/CQRS/LeanCode.CQRS.AspNetCore/ServiceCollectionCQRSExtensions.cs

```diff
@@ -88,4 +89,59 @@ public CQRSServicesBuilder AddCQRSObjects(TypesCatalog contractsCatalog, TypesCa
         objectsSource.AddCQRSObject(CQRSObjectKind.Operation, typeof(TOperation), typeof(TResult), typeof(THandler));
         return this;
     }
+
+    public CQRSServicesBuilder WithLocalCommands(Action<ICQRSApplicationBuilder> configure)
+    {
+        Services.AddSingleton<Local.ILocalCommandExecutor>(
+            s => new Local.MiddlewareBasedLocalCommandExecutor(s, s.GetRequiredService<ICQRSObjectSource>(), configure)
+        );
+        return this;
+    }
+
+    public CQRSServicesBuilder WithLocalQueries(Action<ICQRSApplicationBuilder> configure)
+    {
+        Services.AddSingleton<Local.ILocalQueryExecutor>(
+            s => new Local.MiddlewareBasedLocalQueryExecutor(s, s.GetRequiredService<ICQRSObjectSource>(), configure)
+        );
+        return this;
+    }
+
+    public CQRSServicesBuilder WithLocalOperations(Action<ICQRSApplicationBuilder> configure)
+    {
+        Services.AddSingleton<Local.ILocalOperationExecutor>(
+            s =>
+                new Local.MiddlewareBasedLocalOperationExecutor(s, s.GetRequiredService<ICQRSObjectSource>(), configure)
+        );
+        return this;
+    }
+
+    public CQRSServicesBuilder WithKeyedLocalCommands(object? serviceKey, Action<ICQRSApplicationBuilder> configure)
```

Comment:
If I understand it correctly, the idea is I can have multiple keyed `ILocalExecutor`s (each with different pipeline, each able to run every command) and it's up to the caller to chose which implementation they wan't to resolve.
 
I find a bit strange design decision - I'd expect to have executor which is doing the 'routing' under the hood, but if the latter is harder to achieve than OK.

Reply:
> If I understand it correctly, the idea is I can have multiple keyed ILocalExecutors (each with different pipeline, each able to run every command) and it's up to the caller to chose which implementation they wan't to resolve.

Yes. I also don't think one should use `ILocal*Executor` directly - it should be wrapped in some higher-level class that does some checking.

> I find it a bit strange a design decision

This mostly follows the `IHttpClientFactory` way of doing things.

> I'd expect to have executor which is doing the 'routing' under the hood

I really don't want to force "one true way" right now, as I'm not really sure that "routing" is a correct approach here. Also, I do think there are easier, faster and more ergonomic ways of doing things that one big router (i.e. the first line of this comment).